encoder/src/main/java/openlr/encoder/postprocessor/AlternatePathLrpProcessor.java

```diff
@@ -0,0 +1,102 @@
+    private OpenLREncoderProperties properties;
+
+    private AlternatePathLrpProcessor(OpenLREncoderProperties properties) {
+        this.properties = properties;
+    }
+
+    public static AlternatePathLrpProcessor with(OpenLREncoderProperties properties){
+        return new AlternatePathLrpProcessor(properties);
+    }
+
+    /**
+     *
+     * @param route linked list of road segment
+     * @return list of indices in the linked list where the alternate path with length under threshold starts.
+     * @throws OpenLRProcessingException
+     */
+    private List<Integer> determineNewIntermediatePoints(List<Line> route) throws OpenLRProcessingException {
+        List<Integer> intermediates = new ArrayList<>();
+        SecondShortestRouteChecker checker = SecondShortestRouteChecker.on(route, properties.getAlternatePathRelativeThreshold());
+        for (int index = 1; index < route.size(); ++index) {
+            if (checker.hasValidDeviationBefore(index)) {
+                intermediates.add(index);
+            }
+        }
+        return intermediates;
+    }
+
+    /**
+     * @param route route between the adjacent location reference points(including the road segment of the destination lrp)
+     * @param lrpPositions indices of the road segment where to insert the intermediate points.
+     * @return list of location reference points with intermediate points
+     * @throws OpenLRProcessingException
+     */
+    private List<LocRefPoint> createNewLRPs(List<Line> route, List<Integer> lrpPositions) throws OpenLRProcessingException {
+        List<LocRefPoint> revisedLrpList = new ArrayList<>();
+        LocRefPoint firstPoint = new LocRefPoint(route.subList(0, lrpPositions.get(0)), properties);
+        revisedLrpList.add(firstPoint);
+        for (int index = 0; index < lrpPositions.size() - 1; ++index) {
+            int from = lrpPositions.get(index);
+            int to = lrpPositions.get(index + 1);
+            LocRefPoint intermediatePoint = new LocRefPoint(route.subList(from, to), properties);
+            revisedLrpList.add(intermediatePoint);
+        }
+        int from = lrpPositions.get(lrpPositions.size() - 1);
+        LocRefPoint lastPoint = new LocRefPoint(route.subList(from, route.size()), properties);
+        revisedLrpList.add(lastPoint);
+        return revisedLrpList;
+    }
+
+
+    /**
+     * Generate a linked list of location reference points where No Partial or fully joined  alternate path of length under the given threshold
+     * exist between the road segments of adjacent location reference points
+     * @param lrps linked list of estimated location reference points
+     * @return linked list of revised lrp point linked list with intermediate points
+     * @throws OpenLRProcessingException
+     */
+    public List<LocRefPoint> process(List<LocRefPoint> lrps) throws OpenLRProcessingException {
+
+        if (properties.insertLrpAtAlternatePath()) {
```

Comment:
To make the code better readable I would revert the logic:
- Instead of ask properties.insertLrpAtAlternatePath() and go into the logic you can better ask the if the property is not present and return, the continue the logic:

`if( !properties.insertLrpAtAlternatePath()){ return lrps; }`

Reply:
fixed in 00c70cf1bd4f3e32f47bccb0523ebe37cbbce6e8